commands/role.bee.inc

```diff
@@ -0,0 +1,387 @@
+        $permissions = implode('\' , \'', user_role_permissions($role_array));
+        if (!empty($permissions)) {
+          $permissions = '\''.$permissions.'\'';
+        }
+        $output[] = array(
+          'type' => 'text',
+          'variables' => array(
+            'value' => strtoupper($role) . ': ' . $permissions ,
+          ),
+        );
+        $output[] = array(
+          'type' => 'text',
+          'variables' => array(
+            'value' => '' ,
+          ),
+        );
+      }
+    }
+    return($output);
+  }
+}
+
+/**
+ * Command callback: List all permissions of the modules.
+ */
+function permissions_bee_callback($arguments, $options) {
+  $module_list = module_list();
+  if (!empty($options['module'])) {
+    // If a module has been specified, check it is active, and return error if
+    // not.
+    $module = strtolower($options['module']);
+    if (!array_key_exists($module, $module_list)) {
+      bee_message(bt("The module '!module' is not enabled.", array(
+        '!module' => $module,
+      )), 'error');
+      return;
+    }
+    // Check if the module has permissions defined.
+    $permissions = module_invoke($module, 'permission');
+    if (!empty($permissions)) {
+      // Convert the permissions array to a comma separated list of quoted
+      // permissions.
+      $permissions = implode('\',\'', array_keys($permissions));
+      $permissions = '\'' . $permissions . '\'';
+      // Prepare the command output.
+      $output = array(array(
+        'type' => 'text',
+        'variables' => array(
+          'value' => bt("The module '!module' has the following permissions: !permissions ", array(
+            '!module' => $module,
+            '!permissions' => $permissions,
+          )),
+        ),
+      ),);
+      return $output;
+    }
+    else {
+      // If the module has no defined permissions, return an error.
+      bee_message(bt("The module '!module' has no permissions defined.", array(
+        '!module' => $module,
+      )), 'error');
+      return;
+    }
+  }
+  else {
+    // If no module is defined, show all modules with permissions.
+    $output = array();
+    // Check all active modules for defined permissions.
+    foreach ($module_list as $key => $module) {
+      $permissions = module_invoke($module, 'permission');
+      if (!empty($permissions)) {
+        // If permissions exist for the module, convert the permissions array
+        // to a comma separated list of quoted permissions.
+        $permissions = implode('\',\'', array_keys($permissions));
+        $permissions = '\''.$permissions.'\'';
+        // Prepare the command output.
+        $output[] = array(
+          'type' => 'text',
+          'variables' => array(
+            'value' => strtoupper($module) . ': ' . $permissions ,
+          ),
+        );
+        $output[] = array(
+          'type' => 'text',
+          'variables' => array(
+            'value' => '' ,
+          ),
+        );
```

Comment:
What is the purpose of this? If there is a good reason, please add a comment to explain.